woocommerce/woocommerce-ios
# [Woo POS] M2 UI: Enter POS mode loading screen

Closes: #13143


## Description


Created POS entry loading view. Eventually, it should be visible when originally entering POS or when reloading the empty view. Therefore, I set the visibility conditions to: itemList.loading and tems.count == 0.

## Solution

1. Created `PointOfSaleLoadingView`. Since this loading view is also used in other places, the intention is to make it reusable in other tasks. 
2. Added `Fonts+WooCommercePOS` for reusable font styles taken from Figma
3. Moved loading view from `ItemListView` to `POSDashbordView`, since the view should appear on top of the content. Loading view within `ItemListView` should only appear with pull to refresh logic.
4. Added basic transition animations

Added tests in another PR to avoid too many PR changes: 

## Testing information

1. Open Woo app
6. Select Menu -> Point of Sale Mode
7. Confirm entry loading view appears according to requirements TfaZ4LUkEwEGrxfnEFzvJj-fi-2710_109469
8. Add items to cart, interact with menu, complete the payment, exit pos
9. Confirm that entry loading doesn't appear anymore

## Screenshots


https://github.com/user-attachments/assets/b288b467-2760-4fed-b153-0cfb363d9a14

---
